src/Compilers/Core/Portable/PublicAPI.txt

```diff
@@ -317,10 +317,29 @@ Microsoft.CodeAnalysis.Diagnostics.CompilationStartAnalysisContext.Options.get -
 Microsoft.CodeAnalysis.Diagnostics.CompilationStartAnalysisContext.RegisterSymbolAction(System.Action<Microsoft.CodeAnalysis.Diagnostics.SymbolAnalysisContext> action, params Microsoft.CodeAnalysis.SymbolKind[] symbolKinds) -> void
```

Comment:
@srivatsn @JohnHamby I have addressed the feedback from the design meeting:

1) Wrap all the analysis options passed to the new ctor of CompilationWithAnalyzers within "CompilationWithAnalyzersOptions"
2) Removed GetSupportedDiagnostics, CreateAnalyzerExceptionDiagnostic and ExceptionDiagnostics, which were added in the previous review iteration.
3) Moved ActionCounts and Execution time related APIs into "AnalyzerTelemetry" type within a Telemetry namespace.